### PR TITLE
FishCaughtInfo Event Patch

### PR DESCRIPTION
**Pull request by M3ales**
Adds an immutable event fired when a fish is caught by the player. I haven't been able to test if this also fires when another player catches a fish, but I believe it shouldn't, since it hooks into a write-operation method on the network stream.

Event Description:
> Type:
> `FishingRod`
> Method:
> `FishingRod.pullFishFromWater(int, int, int, int, bool, bool);`
> Type:
> Prefix only - **Immutable**.

### CODE REVIEW

File: Bookcase/Patches/StardewValley/Tools/FishingRod/FishCaughtPatch.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        public Type TargetType => typeof(StardewValley.Tools.FishingRod);
+
+        public MethodInfo TargetMethod => TargetType.GetMethod("pullFishFromWater");
+
+        public static bool Prefix(int whichFish, int fishSize, int fishQuality, int fishDifficulty, bool treasureCaught, bool wasPerfect = false)
+        {
+            AfterFishCaughtEvent caughtEvent = new AfterFishCaughtEvent(whichFish, fishSize, fishQuality, fishDifficulty, treasureCaught, wasPerfect);
+            BookcaseEvents.AfterFishCaught.Post(caughtEvent);
+            return true;
```
Comment (darkhax) [inline Bookcase/Patches/StardewValley/Tools/FishingRod/FishCaughtPatch.cs]: If this patch can't be canceled, you should set return type to void. bool is optional.

File: Bookcase/Events/Tools/FishingRod/AfterFishCaughtEvent.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        public int FishID { get; private set; }
+        /// <summary>
+        /// The size of the fish in inches.
+        /// </summary>
+        public int FishSize { get; private set; }
+        /// <summary>
+        /// The quality of the fish.
+        /// </summary>
+        public int FishQuality { get; private set; }
+        public int FishDifficulty { get; private set; }
```
Comment (darkhax) [inline Bookcase/Events/Tools/FishingRod/AfterFishCaughtEvent.cs]: This one is missing documentation. Also why are they below the constructor?

File: Bookcase/Bookcase.csproj [hunk 3]
```diff
@@ -72,6 +74,7 @@
   <ItemGroup>
     <Analyzer Include="..\packages\Pathoschild.Stardew.ModBuildConfig.2.1.0\analyzers\dotnet\cs\StardewModdingAPI.ModBuildConfig.Analyzer.dll" />
   </ItemGroup>
+  <ItemGroup />
```
Comment (darkhax) [inline Bookcase/Bookcase.csproj]: Is this valid XML?

Comment (M3ales) [reply Bookcase/Bookcase.csproj]: open close tag
<tag/>
But yeah idk why that got added, it's basically a blank itemgroup there.

**Comment by M3ales** [comment]
Kinda messed up the commit chain with a revert earlier because of miss-push of commit meant for different repo. It should be fixed now.

Also addressed @darkhax review.

**Comment by M3ales** [comment]
Removed duplicate.

**Pull request merged**